doc/languages-frameworks/rust.section.md

```diff
@@ -75,6 +75,69 @@ pkgs.rustPlatform.buildRustPackage {
 }
 ```
 
+### Running package tests
+
+When using `buildRustPackage`, the `checkPhase` is enabled by default and runs
+`cargo test` on the package to build. To make sure that we don't compile the
+sources twice and to actually test the artifacts that will be used at runtime, 
+the tests will be ran in the `release` mode by default.
+
+However, in some cases the test-suite of a package doesn't work properly in the
```

Comment:
I think it should be noted explicitly that using `checkType = "debug";` will compile two times -- in release mode for the actual package, and in debug mode for testing (as was happening before this change).

e.g. "NOTE: This will cause two compilations -- one in `release` mode for the package itself, and one in `debug` mode for the tests." or something.